pkg/controller/labelidentity/label_group_index.go

```diff
@@ -299,45 +313,26 @@ func (i *LabelIdentityIndex) deleteSelector(selectorKey string, policyKey string
 	}
 }
 
-// SetPolicySelectors registers ClusterSet-scope policy selectors with the labelIdentityIndex,
-// and then retrieves all the LabelIdentity IDs that currently match these selectors.
-func (i *LabelIdentityIndex) SetPolicySelectors(selectors []*types.GroupSelector, policyKey string) []uint32 {
-	var labelIdentityIDs []uint32
+// SetPolicySelectors registers a policy's latest selectors with the index.
+// For the selectors that the policy no longer possess, it removes the policy key
+// from those selectorItems.
+func (i *LabelIdentityIndex) SetPolicySelectors(selectors []*types.GroupSelector, policyKey string) {
 	newSelectors := map[string]*types.GroupSelector{}
 	for _, s := range selectors {
-		klog.V(4).InfoS("Getting matched LabelIdentity for policy selector", "selector", s.NormalizedName, "policy", policyKey)
 		newSelectors[s.NormalizedName] = s
 	}
 	originalSelectors := i.getPolicySelectors(policyKey)
-	for selKey, sel := range newSelectors {
+	for selKey := range newSelectors {
 		if _, exists := originalSelectors[selKey]; exists {
-			// These clusterset-scoped selectors are already bound to the policy in labelIdentityIndex.
-			// We can simply read matched label identity IDs from the index.
-			selectedLabelIDs := i.GetLabelIdentityIDs(selKey)
-			labelIdentityIDs = append(labelIdentityIDs, selectedLabelIDs...)
-			// Remove matched clusterset-scoped selectors of the policy before and after the update.
-			// The selectors remaining in originalSelectors will need to be removed from the policy.
+			// Remove matched ClusterSet-scoped selectors of the policy before and after the update.
+			// The selectors remaining in originalSelectors will need to be unbounded from the policy.
 			delete(originalSelectors, selKey)
-		} else {
-			selectedLabelIDs := i.AddSelector(sel, policyKey)
-			labelIdentityIDs = append(labelIdentityIDs, selectedLabelIDs...)
 		}
 	}
 	// The policy no longer has these selectors.
 	for selectorKey := range originalSelectors {
 		i.DeleteSelector(selectorKey, policyKey)
```

Comment:
Is it more efficient to delete selectors in a batch, rather than one by one?

Reply:
Not really... policyKey need to be removed from individual selectorItems anyways, so batching does not help.